### PR TITLE
Add crystal-icr v0.2.3 (new formula)

### PR DESCRIPTION
**Pull request by marceloboeira**
Solves: greyblake/crystal-icr#2

### CODE REVIEW

**Comment by greyblake** [comment]
@marceloboeira The test at travis fails:
```
Error: 4 problems in 1 formula
==> audit problems
crystal-icr:
 * `version` (line 7) should be put before `checksum` (line 5)
 * A `test do` test block should be added
 * Stable: version 0.1.0 is redundant with version scanned from URL
 * GitHub repository too new (<30 days old)
```

**Comment by greyblake** [comment]
`GitHub repository too new (<30 days old)` for this we can only wait)

**Comment by marceloboeira** [comment]
@greyblake 

 * `version` (line 7) should be put before `checksum` (line 5): **will do it**
 * A `test do` test block should be added: **Looking for documentation on that**
 * Stable: version 0.1.0 is redundant with version scanned from URL **should we remove the URL?**
 * GitHub repository too new (<30 days old) **How old is the repository?**

**Comment by greyblake** [comment]
> should we remove the URL?

We can try, and see what will CI return.

>  How old is the repository?
Only few days. Basically, we can only wait for 4 weeks)

**Comment by bfontaine** [comment]
> should we remove the URL?

No, remove the version instead. If you remove the URL we can’t download `crystal-icr` anymore :wink:

**Comment by bfontaine** [comment]
> We can try, and see what will CI return.

You can run `brew audit --strict --online crystal-icr` locally; it’s faster and saves CI time for other pull-requests :smiley:

**Comment by marceloboeira** [comment]
@bfontaine thanks, I'll take a look on this problems.

Is there something we could do about the 30 days rule or just wait?

File: Library/Formula/crystal-icr.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "f0189e16f466563b075b26ae71e81572ae120952f8f4b5571bbee41011f471c1"
+
+  depends_on "crystal-lang" => :build
+
+  def install
+    system "make"
+    bin.install "bin/icr"
+  end
+
+  test do
```
Comment (bfontaine) [inline Library/Formula/crystal-icr.rb]: Please write a test in this block.

**Comment by bfontaine** [comment]
> Is there something we could do about the 30 days rule or just wait?

Yep we’ll wait.

> Where can I find docs about testing the formula?

Check the Formula Cookbook: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Formula-Cookbook.md#audit-the-formula

**Comment by greyblake** [comment]
@marceloboeira I've just released icr v0.1.1, it supports crystal 0.11.0 (the new one)

**Comment by marceloboeira** [comment]
@greyblake I'll check that soon as I find documentation on how to test. The link has no examples, because the only think to test is the presence of the bin file on the `/usr/local/bin` path ...

File: Library/Formula/crystal-icr.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class CrystalIcr < Formula
+  desc "Interactive console for Crystal programming language"
+  homepage "https://github.com/greyblake/crystal-icr"
+  url "https://github.com/greyblake/crystal-icr/archive/v0.1.0.tar.gz"
+  sha256 "f0189e16f466563b075b26ae71e81572ae120952f8f4b5571bbee41011f471c1"
+
+  depends_on "crystal-lang" => :build
```
Comment (bfontaine) [inline Library/Formula/crystal-icr.rb]: You can remove `=> :build` because it needs Crystal during runtime, too.

**Comment by bfontaine** [comment]
@greyblake Do you mind tagging a new release so we can use the latest fixes? It’ll also make it easier to write a test.

We should then be able to use something like:
```rb
test do
  assert_equal "=> 4", pipe_output("#{bin}/icr", "2+2")
end
```

**Comment by greyblake** [comment]
@bfontaine see v0.1.2. Thanks)

**Comment by marceloboeira** [comment]
@bfontaine @greyblake I'm trying to test here...

**Comment by marceloboeira** [comment]
Now the only problem is the repository `age` :dancer:

Lets wait!

**Comment by bfontaine** [comment]
Alternatively you might want to maintain the formula in your own tap: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md

**Comment by kepler0** [comment]
I think we're good on age now.

**Comment by marceloboeira** [comment]
@kepler0 yeap, but still not on "popularity"

**Comment by mikemcquaid** [comment]
Please consider [creating your own tap](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-to-Create-and-Maintain-a-Tap.md) instead. Thanks!

**Comment by Sija** [comment]
@MikeMcQuaid Hi, would you consider reopening this PR?

**Comment by UniqMartin** [comment]
@Sija See Homebrew/homebrew-core#1579. This repository is no longer accepting pull requests.

**Comment by Sija** [comment]
@UniqMartin thanks!

**Pull request closed without merging**